openwisp_monitoring/device/apps.py

```diff
@@ -49,6 +53,19 @@ def ready(self):
         self.register_dashboard_items()
         self.register_menu_groups()
         self.add_connection_ignore_notification_reasons()
+        self.connect_check_signals()
+
+    def connect_check_signals(self):
+        post_save.connect(
+            check_post_save_receiver,
```

Comment:
Why not just point it to `DeviceData.handle_unknown_status_change` and have one class method there? Look at the class method receivers used in `connect_device_signals`.